### ISSUE TITLE
Invalid references and unknown fields in twisted.internet

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8531]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)| @acabhishek942 reported|
|-|-|
|Trac ID|trac#8531|
|Type|enhancement|
|Created|2016-06-28 10:16:30Z|
|Branch|https://github.com/twisted/twisted/tree/8531-acabhishek942-doc|

Following are the invalid references and unknown fields in twisted.internet obtained by checking out the trunk and running

```
./bin/admin/build-apidocs . apidocs

```
```
found unknown field on 'twisted.internet.process._FDDetector': <Field 'ivars' 'getpid' [u'The implementatio...']>
found unknown field on 'twisted.internet.process._FDDetector': <Field 'ivars' 'listdir' [u'The implementatio...']>
found unknown field on 'twisted.internet.process._FDDetector': <Field 'ivars' 'openfile' [u'The implementatio...']>
twisted.internet._newtls.ConnectionMixin.TLS: invalid ref to ISSLTransport.providedBy
twisted.internet._newtls.ConnectionMixin: invalid ref to ISSLTransport.providedBy
twisted.internet._sslverify.CertBase.__conform__: invalid ref to Interface
twisted.internet._sslverify.PublicKey.__init__: invalid ref to crypto.PKey
twisted.internet._sslverify.PublicKey: invalid ref to crypto.PKey
twisted.internet.cfreactor.CFReactor._cfrunloop: invalid ref to CFRunLoop
twisted.internet.cfreactor.CFReactor._inCFLoop: invalid ref to CFRunLoopRun
twisted.internet.cfreactor.CFReactor._socketCallback: invalid ref to CFSocket
twisted.internet.cfreactor.CFReactor._socketCallback: invalid ref to CFSocketCreateWithNative
twisted.internet.cfreactor.CFReactor._socketCallback: invalid ref to kCFSocketReadCallBack
twisted.internet.cfreactor.CFReactor._socketCallback: invalid ref to kCFSocketWriteCallBack
twisted.internet.cfreactor.CFReactor._unwatchFD: invalid ref to CFRunLoop
twisted.internet.cfreactor.CFReactor._unwatchFD: invalid ref to kCFSocketReadCallBack
twisted.internet.cfreactor.CFReactor._unwatchFD: invalid ref to kCFSocketWriteCallBack
twisted.internet.cfreactor.CFReactor._watchFD: invalid ref to CFRunLoop
twisted.internet.cfreactor.CFReactor._watchFD: invalid ref to kCFSocketReadCallBack
twisted.internet.cfreactor.CFReactor._watchFD: invalid ref to kCFSocketWriteCallBack
twisted.internet.cfreactor.CFReactor.mainLoop: invalid ref to CFRunLoopRun
twisted.internet.cfreactor.CFReactor: invalid ref to CFRunLoop
twisted.internet.cfreactor.CFReactor: invalid ref to CFRunLoopRun
twisted.internet.cfreactor.install: invalid ref to CFRunLoopRun
twisted.internet.defer.Deferred._continuation: invalid ref to _addContinue
twisted.internet.defer.Deferred._continuation: invalid ref to _continue
twisted.internet.defer.Deferred._continuation: invalid ref to _removeContinue
twisted.internet.defer.DeferredList.__init__: invalid ref to Deferreds
twisted.internet.defer.TimeoutError: invalid ref to Deferred.setTimeout
twisted.internet.defer._CONTINUE: invalid ref to Deferred.callbacks
twisted.internet.defer.gatherResults: invalid ref to Deferreds
twisted.internet.defer: invalid ref to Deferred.callbacks
twisted.internet.endpoints._parseUNIX: invalid ref to IReactorTCP.listenUNIX
twisted.internet.endpoints._tokenize: invalid ref to _OP
twisted.internet.endpoints._tokenize: invalid ref to _STRING
twisted.internet.iocpreactor.tcp.Connection.loseConnection: invalid ref to ITCPTransport.loseConnection
twisted.internet.iocpreactor.tcp.Connection.write: invalid ref to ITCPTransport.write
twisted.internet.iocpreactor.tcp.Connection.writeSequence: invalid ref to ITCPTransport.writeSequence
twisted.internet.win32eventreactor: invalid ref to Process
```

<details><summary>Searchable metadata</summary>

```
trac-id__8531 8531
type__enhancement enhancement
reporter__acabhishek942 acabhishek942
priority__normal normal
milestone__None None
branch__8531_acabhishek942_doc 8531-acabhishek942-doc
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1467108990676952 1467108990676952
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_1"></a>|@acabhishek942 commented|
|-|-|

See: https://github.com/twisted/twisted/pull/258

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 1a36a00cd6ef2eedbd5693b1da0e088df61b7c47
```
#!CommitTicketReference repository="" revision="1a36a00cd6ef2eedbd5693b1da0e088df61b7c47"
Merge 8531-acabhishek942-doc: Fix invalid references and unknown fields in twisted.internet

Author: acabhishek942
Reviewer: rodrigc
Fixes: #8531
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
